File: TwoSum/TwoSum.java
package TwoSum;

import java.util.HashMap;

public class TwoSum {
	public int[] twoSum(int[] numbers, int target) {
		int len = numbers.length;
		int[] index = new int[2];

		HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();

		for (int i = 0; i < len; i++) {
			if (!map.containsKey(numbers[i])) {
				map.put(target - numbers[i], i);
			} else {
				index[0] = map.get(numbers[i]);
				index[1] = i;
			}
		}

		return index;
	}

	public static void main(String[] args) {
		int[] a = {3,2,4,3,3};
		int[] b = new TwoSum().twoSum(a, 6);
		System.out.println(b[0] + " " + b[1]);
	}
}

File: TwoSum/twosum.py
#!/usr/bin/python
#coding=utf-8

class Solution(object):
	def twoSum(self, nums, target):
		"""
		:type nums: List[int]
		:type target: int
		:rtype: List[int]
		"""
		for x in nums:
			left = target - x
			if left in nums:
				try:
					if left == x:
						index_a = nums.index(x)
						index_b = nums.index(x, index_a + 1)
						return [index_a, index_b]
				except:
					continue
				return [nums.index(x),nums.index(left)]

a = [3,2,4,3]
print Solution().twoSum(a, 6)
